arch/riscv/core/fatal.c

```diff
@@ -28,6 +28,84 @@ static const struct z_exc_handle exceptions[] = {
  #define NO_REG "                "
 #endif
 
+#ifdef CONFIG_RISCV_EXCEPTION_STACK_TRACE
+#define MAX_STACK_FRAMES 8
+
+struct stackframe {
+	uintptr_t fp;
+	uintptr_t ra;
+};
+
+static bool in_stack_bound(uintptr_t addr)
+{
+#ifdef CONFIG_THREAD_STACK_INFO
+	uintptr_t start, end;
+
+	if (_current == NULL || arch_is_in_isr()) {
+		/* We were servicing an interrupt */
+		int cpu_id;
+
+#ifdef CONFIG_SMP
+		cpu_id = arch_curr_cpu()->id;
+#else
+		cpu_id = 0;
+#endif
+
+		start = (uintptr_t)Z_THREAD_STACK_BUFFER(z_interrupt_stacks[cpu_id]);
+		end = start + CONFIG_ISR_STACK_SIZE;
+#ifdef CONFIG_USERSPACE
+		/* TODO: handle user threads */
+#endif
+	} else {
+		start = _current->stack_info.start;
+		end = Z_STACK_PTR_ALIGN(_current->stack_info.start + _current->stack_info.size);
+	}
+
+	return (addr >= start) && (addr < end);
+#else
+	ARG_UNUSED(addr);
+	return true;
+#endif /* CONFIG_THREAD_STACK_INFO */
+}
+
+static inline bool in_text_region(uintptr_t addr)
+{
+#ifndef CONFIG_SOC_ESP32C3
+	extern uintptr_t __text_region_start, __text_region_end;
```

Comment:
@marekmatej is there a `__text_region_start` & `__text_region_end` equivalent for esp32c3?

Reply:
probably `_text_start` & `_text_end`?

Reply:
Judging by `build/zephyr/zephyr.map` that looks like it should be the case

Reply:
Created #71363 for this